lifecycle-operator/controllers/lifecycle/keptntask/job_utils.go

```diff
@@ -76,6 +76,8 @@ func (r *KeptnTaskReconciler) getJob(ctx context.Context, jobName string, namesp
 }
 
 func (r *KeptnTaskReconciler) generateJob(ctx context.Context, task *klcv1alpha3.KeptnTask, definition *klcv1alpha3.KeptnTaskDefinition, request ctrl.Request) (*batchv1.Job, error) {
+	serviceAccountName := getServiceAccount(definition.Spec.ServiceAccount)
+	automountServiceAccountToken := getAutomountServiceAccountToken(definition.Spec.AutomountServiceAccountToken)
```

Comment:
Let's make these two functions the struct functions, what I mean is:

```
func (d *KeptnTaskDefinition) GetServiceAccount() {
	if d.Spec.ServiceAccount == nil {
		return ""
	}
	return serviceAccount.Name
}
```
for both functions

Reply:
hey @odubajDT, where do you mean to put the function, if the objective is in the same file,  `(d *KeptnTaskDefinition)` cannot be used 

![kkkk)](https://github.com/keptn/lifecycle-toolkit/assets/98270250/2a27d1ea-7bbd-45c2-b3ee-a97a1d25fc8b)

Reply:
Hello @prakrit55 , we should put this methods to this file https://github.com/keptn/lifecycle-toolkit/blob/main/lifecycle-operator/apis/lifecycle/v1alpha3/keptntaskdefinition_types.go

Reply:
Previously, I guessed that one but then the lint was throwing function unused errors 😊. Thanks, its working.

Reply:
> Let's make these two functions the struct functions, what I mean is:
> 
> ```
> func (d *KeptnTaskDefinition) GetServiceAccount() string {
> 	if d.Spec.ServiceAccount == nil {
> 		return ""
> 	}
> 	return serviceAccount.Name
> }
> ```
> 
> for both functions
> 
> Also please provide some unit tests

hii @odubajDT, I am still new to unit tests some hint wd work

Reply:
Sure, first try to look at the official Go documentation about writing tests https://go.dev/doc/tutorial/add-a-test

The you can look at already existing unit tests we have in api package https://github.com/keptn/lifecycle-toolkit/blob/main/lifecycle-operator/apis/lifecycle/v1alpha3/keptnappcreationrequest_types_test.go

which are testing the functionality of these functions https://github.com/keptn/lifecycle-toolkit/blob/main/lifecycle-operator/apis/lifecycle/v1alpha3/keptnappcreationrequest_types.go

Something similar is required from you as well